### PR TITLE
Nlperspectives ted

### PR DESCRIPTION
**Pull request by wojtek11530**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
+## To Do:
+
+- a preprocessing step of averaging the embedding with attention mask > 0 before feeding into fc1
+
+## Questions:
+When I try to use some code from embeddings.py to average the embedding with attention mask > 0, (i.e. the line "mask = batch_encoding attention_mask > 0" and the loop "for i in range(emb....):....", I get the error saying "there are too many indices for tensor of dimension 1".
+
+I tried changing the line "all_embeddings.append(emb\[0\]\[i, mask\[i\] > 0, :\].mean(axis=0)\[None, :\])" into "all_embeddings.append(emb\[0\]\[mask\[i\]].mean(axis=0)\[None, :\])" and I get another error saying that the mask's shape \[128\] does not match the tensor's shape \[768\].
+
+How to do this correctly?
```
Comment (wojtek11530) [inline README.md:29]: Hmm, I have not met that problem so I did not toch that code. Do you still have it? If I understand right, you couldn't get the pretrained embeddings for texts. I think, it would be easier if you show me that during some debugging.

**Comment by wojtek11530** [comment]
We are not merging it so far. When it will be ready (soon or later :p ), it will demand to squash commits.

File: README.md (hunk 1 above)
Comment (teddy-f-47) [reply README.md:29]: Sorry for not updating the readme; I have deleted that piece of code. I wrote that question when Anh told me that I should "check the /utils/embedding.py, in function _get_embeddings() before concatenating all the embeddings, there's a preprocessing step of averaging the embedding with attention mask > 0. All of the other text embeddings we create for other models (using data module) are applied to this processing. Therefore, I think we can add that additional part also, before concat them and feeding to fc1."

From what I understand, I am supposed to do the averaging of the embeddings with attention mask > 0 before feeding them to fc1. But I was not able to do that, as I kept getting errors that I didn't really understand 😓 . So I deleted the piece of code for averaging the embeddings with attention mask > 0, and the model seemed to run fine. In your opinion, should I implement this averaging pre-processing?

Comment (wojtek11530) [reply README.md:29]: I have debugged it and the resulting avereged emebddings looks the same in both verions:
image

So if your version is the same as for that `avg1` in the screenshot, it is ok. Still, I don't know why it is not working for ```emb[0][i, mask[i] > 0, :].mean(axis=0)[None, :]```. If you want you can check if you have the same error for running experiment for jester.

File: personalized_nlp/learning/train_alt.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        progress_bar_refresh_rate=20,
+        log_every_n_steps=10,
+        logger=logger,
+        callbacks=callbacks,
+        **trainer_kwargs,
+    )
+    trainer.fit(model, train_loader, val_loader)
+    
+    if flag_run_test == True:
+        trainer.test(test_dataloaders=test_loader)
```
[the remark this answers is not in the document]
Comment (wojtek11530) [reply personalized_nlp/learning/train_alt.py]: Yeah, your solution is very nice.
